lms/djangoapps/django_comment_client/base/tests.py

```diff
@@ -866,6 +911,42 @@ def un_flag_comment(self, mock_request, is_closed):
 
         assert_equal(response.status_code, 200)
 
+    @ddt.data(
+        ('upvote_thread', 'thread_id', thread_voted),
+        ('upvote_comment', 'comment_id', comment_voted),
+        ('downvote_thread', 'thread_id', thread_voted),
+        ('downvote_comment', 'comment_id', comment_voted)
+    )
+    @ddt.unpack
+    def test_voting(self, view_name, item_id, signal, mock_request):
+        self._setup_mock_request(mock_request)
+        self.setup_signal_handler(signal)
+
+        response = self.client.post(
+            reverse(
+                view_name,
+                kwargs={item_id: 'dummy', 'course_id': self.course_id.to_deprecated_string()}
+            )
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assert_signal_sent(signal, sender=None, user=self.student)
+
+    def test_endorse_comment(self, mock_request):
+        self._setup_mock_request(mock_request)
+        self.setup_signal_handler(comment_endorsed)
+        self.client.login(username=self.moderator.username, password=self.password)
```

Comment:
Comments can only be endorsed by discussion staff/moderators/TAs, or the student who posted the question. Since these tests are creating dummy questions, it's easier to test with a moderator.